app/models/user.rb

```diff
@@ -235,19 +235,20 @@ def display_official_position_badge?
   end
 
   def block
-    debates_ids = Debate.where(author_id: id).pluck(:id)
-    comments_ids = Comment.where(user_id: id).pluck(:id)
-    proposal_ids = Proposal.where(author_id: id).pluck(:id)
-    investment_ids = Budget::Investment.where(author_id: id).pluck(:id)
-    proposal_notification_ids = ProposalNotification.where(author_id: id).pluck(:id)
-
     hide
 
-    Debate.hide_all debates_ids
-    Comment.hide_all comments_ids
+    Debate.hide_all debate_ids
+    Comment.hide_all comment_ids
     Proposal.hide_all proposal_ids
-    Budget::Investment.hide_all investment_ids
-    ProposalNotification.hide_all proposal_notification_ids
+    Budget::Investment.hide_all budget_investment_ids
+    ProposalNotification.hide_all ProposalNotification.where(author_id: id).pluck(:id)
+  end
+
+  def after_restore
+    Debate.restore_all debate_ids
```

Comment:
@carlosiniesta Sorry, I was editing the messages in two places at the same time and messed up with this comment pray.

IMHO we shouldn't restore records which were already hidden before hiding the user :thinking:. This is tricky, though :thinking:.

Reply:
That make a lot of sense. I did it, now only related content hidden at the same time will be restored.